File: source/code/vbnet/authentication/create_directory_linkedin.vb
Imports Stormpath.SDK.Provider

Dim linkedInDirectory = client.Instantiate(Of IDirectory)()
linkedInDirectory.SetName("My LinkedIn Directory")

Await client.CreateDirectoryAsync(
    linkedInDirectory,
    Function(options) options.ForProvider(client _
                                          .Providers() _
                                          .LinkedIn() _
                                          .Builder() _
                                          .SetClientId("YOUR_LINKEDIN_APP_ID") _
                                          .SetClientSecret("YOUR_LINKEDIN_APP_SECRET") _
                                          .Build()))

File: source/code/nodejs/authentication/create_directory_linkedin.js
var linkedInDirectory = {
  name : 'my-linkedin-directory',
  description : 'A LinkedIn directory',
  provider: {
    providerId: 'linkedin',
    clientId: 'YOUR_LINKEDIN_APP_ID',
    clientSecret: 'YOUR_LINKEDIN_APP_SECRET'
  }
};

client.createDirectory(linkedInDirectory, function (err, directory) {
  if (err) {
    return console.error(err);
  }

  console.log('LinkedIn directory created!');
});